domain/src/main/java/org/oppia/android/domain/feedbackreporting/FeedbackReportManagementController.kt

```diff
@@ -0,0 +1,346 @@
+  private fun getDeviceContext(
+    deviceContext: FeedbackReportingDeviceContext
+  ): GaeFeedbackReportingDeviceContext {
+    return GaeFeedbackReportingDeviceContext(
+      deviceModel = deviceContext.deviceModel,
+      sdkVersion = deviceContext.sdkVersion,
+      buildFingerprint = deviceContext.buildFingerprint,
+      networkType = deviceContext.networkType.name.toLowerCase()
+    )
+  }
+
+  // Helper function that construct a Moshi data class object for the app's information.
+  private fun getAppContext(
+    appContext: FeedbackReportingAppContext
+  ): GaeFeedbackReportingAppContext {
+    return GaeFeedbackReportingAppContext(
+      entryPoint = getEntryPointData(appContext),
+      textSize = appContext.textSize.name.toLowerCase(),
+      textLanguageCode = appContext.textLanguage.toLanguageCode().toLanguageCodeString(),
+      audioLanguageCode = appContext.audioLanguage.toLanguageCode().toLanguageCodeString(),
+      downloadAndUpdateOnlyOnWifi = appContext.deviceSettings.allowDownloadAndUpdateOnlyOnWifi,
+      automaticallyUpdateTopics = appContext.deviceSettings.automaticallyUpdateTopics,
+      isAdmin = appContext.isAdmin,
+      eventLogs = eventsLogList ?: emptyList(),
+      logcatLogs = getLogcatLogs(),
+    )
+  }
+
+  // Helper function that construct a Moshi data class object for the entry-point used by the user.
+  private fun getEntryPointData(
+    appContext: FeedbackReportingAppContext
+  ): GaeFeedbackReportingEntryPoint {
+    var topicId: String? = null
+    var storyId: String? = null
+    var explorationId: String? = null
+    var subtopicId: String? = null
+    when (appContext.entryPointCase) {
+      // Get the current topic information if the report is sent during a lesson or revision session.
+      EntryPointCase.LESSON_PLAYER -> {
+        val lesson = appContext.lessonPlayer
+        topicId = lesson.topicId
+        storyId = lesson.storyId
+        explorationId = lesson.explorationId
+      }
+      EntryPointCase.REVISION_CARD -> {
+        val revisionCard = appContext.revisionCard
+        topicId = revisionCard.topicId
+        subtopicId = revisionCard.subtopicId
+      }
+      EntryPointCase.NAVIGATION_DRAWER -> {
+        // If entry point is not an exploration player or revision card, leave story values as null.
+      }
+      EntryPointCase.CRASH_DIALOG -> {
+        // If entry point is not an exploration player or revision card, leave story values as null.
+      }
+      else -> throw IllegalArgumentException(
+        "Encountered unexpected entry point: ${appContext.entryPointCase.name}"
+      )
+    }
+    return GaeFeedbackReportingEntryPoint(
+      entryPointName = appContext.entryPointCase.name.toLowerCase(),
+      topicId = topicId,
+      storyId = storyId,
+      explorationId = explorationId,
+      subtopicId = subtopicId
+    )
+  }
+
+  // Helper function to retrieve the logcat logs from the device.
+  private fun getLogcatLogs(): List<String> {
```

Comment:
I suggest testing multiple logcat lines (unless I'm mistaken, I think your current test data only covers a single line which might miss some of the logic being employed here).